code/game/gamemodes/blob/theblob.dm

```diff
@@ -196,6 +198,11 @@
 		color = a_color
 	return
 
+/obj/effect/blob/proc/update_desc(var/a_desc=null)

```

Comment:
I think that would be more sane. Just set the desc whenever the blob part reagent is first added. Then it'll just be in one place.